File: LeetCode Solutions/LeetCoding Challenge/May 2020 LeetCoding Challenge/Jump Game II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        numsLength, current, Next, result, i = len(nums) - 1, -1, 0, 0, 0
        while(Next < numsLength):
            if(i > current):
                result += 1
                current = Next
            Next = max(Next, nums[i] + i)
            i += 1
        return result

File: LeetCode Solutions/LeetCoding Challenge/May 2020 LeetCoding Challenge/Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        int numsLength = nums.length - 1, current = -1, next = 0, result = 0;
        for(int i=0; next<numsLength; i++){
            if(i > current){
                result++;
                current = next;
            };
            next = Math.max(next, nums[i] + i);
        };
        return result;
    }
}